x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/action/ListEnginesAction.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.entsearch.engine.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.StatusToXContentObject;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.core.action.util.PageParams;
+import org.elasticsearch.xpack.entsearch.engine.Engine;
+import org.elasticsearch.xpack.entsearch.engine.EngineIndexService;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class ListEnginesAction extends ActionType<ListEnginesAction.Response> {
```

Comment:
Nitpick: We called this the List Engines call in the Rails API. However, would it make sense to call this `SearchEnginesAction` instead, since we're effectively searching every call?

Reply:
I think List describes the action accurately, but I'm open to a name change.

Reply:
I don't have really strong feelings about it, it's OK to keep as list.